alerter/src/main/java/org/apache/hertzbeat/alert/service/impl/AlertDefineServiceImpl.java

```diff
@@ -167,30 +185,20 @@ public Page<AlertDefine> getAlertDefines(List<Long> defineIds, String search, By
                 }
                 andList.add(inPredicate);
             }
-            if (StringUtils.hasText(search)) {
-                Predicate predicate = criteriaBuilder.or(
-                        criteriaBuilder.like(
-                                criteriaBuilder.lower(root.get("app")),
-                                "%" + search.toLowerCase() + "%"
-                        ),
-                        criteriaBuilder.like(
-                                criteriaBuilder.lower(root.get("metric")),
-                                "%" + search.toLowerCase() + "%"
-                        ),
-                        criteriaBuilder.like(
-                                criteriaBuilder.lower(root.get("field")),
-                                "%" + search.toLowerCase() + "%"
-                        ),
-                        criteriaBuilder.like(
-                                criteriaBuilder.lower(root.get("expr")),
-                                "%" + search.toLowerCase() + "%"
-                        ),
-                        criteriaBuilder.like(
-                                criteriaBuilder.lower(root.get("template")),
-                                "%" + search.toLowerCase() + "%"
-                        )
-                );
-                andList.add(predicate);
+            if (null != finalSearchList && !finalSearchList.isEmpty()) {
+                List<Predicate> searchPredicates = new ArrayList<>();
+                for (String searchContent : finalSearchList) {
+                    Predicate predicate = criteriaBuilder.or(
+                            criteriaBuilder.like(criteriaBuilder.lower(root.get("app")), "%" + searchContent.toLowerCase() + "%"),
```

Comment:
set all  finalSearchList content toLowerCase(), to  reduce searchContent.toLowerCase()